bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java

```diff
@@ -262,32 +262,7 @@ private void handleBridgeChildStatusUpdate(Thing thing, ThingStatusInfo oldStatu
 
         @Override
         public void thingUpdated(final Thing thing) {
-            thingUpdatedLock.add(thing.getUID());
-            AccessController.doPrivileged((PrivilegedAction<@Nullable Void>) () -> {
-                Provider<Thing> provider = thingRegistry.getProvider(thing);
-                if (provider == null) {
-                    throw new IllegalArgumentException(MessageFormat.format(
-                            "Provider for thing {0} cannot be determined because it is not known to the registry",
-                            thing.getUID().getAsString()));
-                }
-                if (provider instanceof ManagedProvider) {
-                    @SuppressWarnings("unchecked")
-                    ManagedProvider<Thing, ThingUID> managedProvider = (ManagedProvider<Thing, ThingUID>) provider;
-                    managedProvider.update(thing);
-                } else {
-                    logger.debug("Only updating thing {} in the registry because provider {} is not managed.",
-                            thing.getUID().getAsString(), provider);
-                    Thing oldThing = thingRegistry.get(thing.getUID());
-                    if (oldThing == null) {
-                        throw new IllegalArgumentException(
-                                MessageFormat.format("Cannot update thing {0} because it is not known to the registry",
-                                        thing.getUID().getAsString()));
-                    }
-                    thingRegistry.updated(provider, oldThing, thing);
-                }
-                return null;
```

Comment:
In fact this contradicts the description in the `ThingHandlerCallback` interface:

```
    /**
     * Informs about an update of the whole thing.
     *
     * @param thing thing that was updated (must not be null)
     * @throws IllegalStateException if the {@link Thing} is read-only.
     */
    void thingUpdated(Thing thing);
```

However, if we strictly follow the interface description this makes any dynamic properties or updates to file-based things impossible. IMO the interface desciption should be updated to reflect the current code.

I also wonder why we check if the thing exists. I would be very astonished if we got a provider from the `ThingRegistry` when the thing itself is not in the registry. I think this was introduced in #1134 to prevent the null-error for `Registry.updated`. It would probably be better to check if the thing is present in the registry before we try to get the provider and fail early if it is not in the registry (the registry fails silently if you try to update a thing that is not in the registry).